07.ls_object/lib/screen.rb

```diff
@@ -18,23 +22,28 @@ def show
     output_blocks << create_output_block_with_file_stats
     output_blocks << create_output_blocks_with_dir_stats
 
-    puts output_blocks.reject(&:empty?).join("\n\n")
+    output_blocks.reject(&:empty?).join("\n\n").encode('UTF-8')
   end
 
   private
 
   def create_output_block_with_file_stats
-    @long_format ? DetailFormatter.new(@stats_in_dir['']).write : NormalFormatter.new(@stats_in_dir['']).write
+    formatter_factory.new(@stats['']).write
   end
 
   def create_output_blocks_with_dir_stats
-    @stats_in_dir.except('').map do |dir_path, stats|
+    @stats.except('').map do |dir_path, stat|
```

Comment:
```suggestion
    @stats.except('').map do |dir_path, stats|
```

ここは配列が入っているのでは？もしそうなら複数形の方が適切ですね

違和感があるなら、`@stats` 側の命名に問題があるのかも